destination/destination_integration_test.go

```diff
@@ -0,0 +1,361 @@
+	is := is.New(t)
+
+	cfg := prepareConfig(t)
+
+	destination := NewDestination()
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	err := destination.Configure(ctx, cfg)
+	is.NoErr(err)
+
+	err = destination.Open(ctx)
+	is.NoErr(err)
+
+	col, err := getTestCollection(ctx, cfg[config.KeyCollection])
+	is.NoErr(err)
+
+	t.Cleanup(func() {
+		err = col.Drop(context.Background())
+		is.NoErr(err)
+
+		err = destination.Teardown(ctx)
+		is.NoErr(err)
+	})
+
+	testItem := createTestItem(t)
+
+	n, err := destination.Write(ctx, []sdk.Record{createTestCreateRecord(t, testItem)})
+	is.NoErr(err)
+	is.Equal(n, 1)
+
+	_, err = destination.Write(ctx, []sdk.Record{createTestUpdateRecordNoKeys(t)})
+	is.True(errors.Is(err, writer.ErrEmptyKey))
+
+	_, err = col.DeleteMany(ctx, bson.M{})
+	is.NoErr(err)
+}
+
+func TestDestination_Write_deleteSuccess(t *testing.T) {
+	is := is.New(t)
+
+	cfg := prepareConfig(t)
+
+	destination := NewDestination()
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	err := destination.Configure(ctx, cfg)
+	is.NoErr(err)
+
+	err = destination.Open(ctx)
+	is.NoErr(err)
+
+	col, err := getTestCollection(ctx, cfg[config.KeyCollection])
+	is.NoErr(err)
+
+	t.Cleanup(func() {
+		err = col.Drop(context.Background())
+		is.NoErr(err)
+
+		err = destination.Teardown(ctx)
+		is.NoErr(err)
+	})
+
+	testItem := createTestItem(t)
+
+	n, err := destination.Write(ctx, []sdk.Record{createTestCreateRecord(t, testItem)})
+	is.NoErr(err)
+	is.Equal(n, 1)
+
+	n, err = destination.Write(ctx, []sdk.Record{createTestDeleteRecord(t, testItem)})
+	is.NoErr(err)
+	is.Equal(n, 1)
+
+	res, err := col.Find(ctx, bson.M{})
+	is.NoErr(err)
+	is.True(!res.Next(ctx))
+
+	_, err = col.DeleteMany(ctx, bson.M{})
+	is.NoErr(err)
+}
+
+func compareTestPayload(
+	ctx context.Context,
+	t *testing.T,
+	is *is.I,
+	col *mongo.Collection,
+	testRecordPayload sdk.StructuredData,
+) {
+	t.Helper()
+
+	c, err := col.CountDocuments(ctx, bson.D{})
+	is.NoErr(err)
+	is.Equal(c, int64(1))
+
+	res, err := col.Find(ctx, bson.M{})
+	is.NoErr(err)
+	is.True(res.TryNext(ctx))
+	var result map[string]any
+	err = res.Decode(&result)
+	is.NoErr(err)
+
+	is.Equal(sdk.StructuredData(result), testRecordPayload)
+}
+
+func getTestCollection(ctx context.Context, collection string) (*mongo.Collection, error) {
+	conn, err := mongo.Connect(ctx, options.Client().ApplyURI(testURI))
+	if err != nil {
+		return nil, fmt.Errorf("connect to mongo: %w", err)
+	}
+
+	return conn.Database(testDB).Collection(collection), nil
+}
+
+func createTestItem(t *testing.T) map[string]any {
+	t.Helper()
+
+	return map[string]any{
+		// testIDFieldName is declared as string for testing codec
+		testIDFieldName:    primitive.NewObjectIDFromTimestamp(time.Now()).String(),
+		testEmailFieldName: gofakeit.Email(),
+		testNameFieldName:  gofakeit.Name(),
+	}
+}
+
+func createTestCreateRecord(t *testing.T, item map[string]any) sdk.Record {
```

Comment:
we are using it for 4 times at this tests (unlike createTestSnapshotRecord etc.), maybe we should not delete it? What do you think about it?

Reply:
I understand your point but this function is not really needed because it is consist only 1 line logic it is better to use 
```
sdk.Util.Source.NewRecordCreate(
		nil, nil,
		// in insert keys are not used, so we can omit it
		nil,
		sdk.StructuredData(item),
	)
```
without function

Reply:
fixed